tasks/sep/ink-001-fp-holocene-pectra-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,233 @@
+        uint256 trackedSignersWithNoCode;
+        address[] memory safeSignersWithNoCode = new address[](numberOfSafeSignersWithNoCode);
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            if (securityCouncilSafeOwners[i].code.length == 0) {
+                safeSignersWithNoCode[trackedSignersWithNoCode] = securityCouncilSafeOwners[i];
+                trackedSignersWithNoCode++;
+            }
+        }
+
+        // Here we add the standard (non Safe signer) exceptions.
+        address[] memory shouldHaveCodeExceptions = new address[](0 + numberOfSafeSignersWithNoCode);
+
+
+        // And finally, we append the Safe signer exceptions.
+        for (uint256 i = 0; i < safeSignersWithNoCode.length; i++) {
+            shouldHaveCodeExceptions[0 + i] = safeSignersWithNoCode[i];
+        }
+
+        return shouldHaveCodeExceptions;
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](5);
+        allowed[0] = address(dgfProxy);
+        allowed[1] = address(ownerSafe);
+        allowed[2] = address(securityCouncilSafe);
+        allowed[3] = address(fndSafe);
+        allowed[4] = livenessGuard;
+    }
+
+    function _postCheck(Vm.AccountAccess[] memory accesses, Simulation.Payload memory) internal view override {
+        console.log("Running post-deploy assertions");
+        checkStateDiff(accesses);
+        checkDGFProxyAndGames();
+        checkMips();
+        checkPermissionedDisputeGame();
+        checkFaultDisputeGame();
+        console.log("All assertions passed!");
+    }
+
+    function checkDGFProxyAndGames() internal view {
+        console.log("check dispute game implementations");
+        require(address(faultDisputeGame) == address(dgfProxy.gameImpls(GameTypes.CANNON)), "dgf-100");
+        require(address(permissionedDisputeGame) == address(dgfProxy.gameImpls(GameTypes.PERMISSIONED_CANNON)), "dgf-200");
+
+        require(faultDisputeGame.version().eq(gameVersion), "game-100");
+        require(permissionedDisputeGame.version().eq(gameVersion), "game-200");
+
+        require(faultDisputeGame.absolutePrestate().raw() == absolutePrestate, "game-300");
+        require(permissionedDisputeGame.absolutePrestate().raw() == absolutePrestate, "game-400");
+
+        require(address(faultDisputeGame.vm()) == newMips, "game-500");
+        require(address(permissionedDisputeGame.vm()) == newMips, "game-600");
+
+        require(faultDisputeGame.l2ChainId() == chainId, "game-700");
+        require(permissionedDisputeGame.l2ChainId() == chainId, "game-800");
+    }
+
+    /// @notice Checks that the FaultDisputeGame was handled correctly.
+    function checkFaultDisputeGame() internal view {
+        // Check that the FaultDisputeGame version is correct.
+        require(LibString.eq(expectedFaultDisputeGameImpl.version(), "1.3.1"), "checkFaultDisputeGame-21");
```

Comment:
Can we also check the addresses of the `DelayedWETH` and `AnchorStateRegistry` references?

Reply:
Okay so we added checks for the `DelayedWETH`, but do we need the AnchorStateRegistry version checked too? It should be changed in this upgrade